### ISSUE TITLE
How to add a custom widget to Builder?

### ISSUE THREAD

**Issue by fryedrycestyle** [rainlab/builder-plugin#105]
Hey there,

I'm a bit new to working with Builder, but the visual components of it are really nice in streamlining plug-in development. I'm only having trouble with 'one' thing.

Adding a new control widget to a model. I noticed this thread: https://github.com/rainlab/builder-plugin/issues/60 but I'm not coming away with much from it. From what I understand something (not sure if it's the widget I built or a separate script) needs to extend the Provider class mentioned.

So basically, what is the process for adding a custom control widget to the Builder widgets menu?

Thanks!

**Comment by daftspunk** [comment]
It might be a good idea to post your plugin on GitHub, that way others will be able to help you with it. The issue #60 does show a clear example on how to add a custom widget to the menu, it might just be a matter of filling a few blank spaces in your implementation.

**Comment by fryedrycestyle** [comment]
So I made some progress, however I'm faced with an error - and it makes me wonder if my path for the partial I've made is correct?

The error is saying it cannot find the partial on line 65 of ViewMaker.php. I quickly looked to see what that was and it had to do with the path of the plugin. I was under the impression partials for a plugin would go in a partials folder under the widgets? I'm probably mistaken here, so I just would like to clarify where my partial should live?
